### PR TITLE
Make the autosklearn PEP8 compliant

### PR DESCRIPTION
**Pull request by gui-miotto**

### CODE REVIEW

File: autosklearn/__init__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import sys
 
 from autosklearn.util import dependencies
-from autosklearn.__version__ import __version__
+from autosklearn.__version__ import __version__  # noqa (imported but unused)
```
Comment (gui-miotto) [inline autosklearn/__init__.py:6]: Is this import really necessary?

File: autosklearn/smbo.py [hunk 2]
```diff
@@ -366,7 +366,7 @@ def run_smbo(self):
         # first create a scenario
         seed = self.seed
         self.config_space.seed(seed)
-        num_params = len(self.config_space.get_hyperparameters())
+        self.config_space.get_hyperparameters()
```
Comment (gui-miotto) [inline autosklearn/smbo.py:369]: I removed the variable assignment, but shouldn't we get rid of the line altogether?

File: autosklearn/evaluation/train_evaluator.py [hunk 3]
```diff
@@ -571,7 +572,7 @@ def fit_predict_and_loss(self, iterative=False):
                 Y_test_pred = None
 
             self.Y_optimization = Y_targets
-            loss = self._loss(Y_targets, Y_optimization_pred)
+            self._loss(Y_targets, Y_optimization_pred)
```
Comment (gui-miotto) [inline autosklearn/evaluation/train_evaluator.py:575]: (again) I removed the variable assignment, but shouldn't we get rid of the line altogether?

File: autosklearn/__init__.py (hunk 1 above)
Comment (mfeurer) [reply autosklearn/__init__.py:6]: Yes, it allows `print(autosklearn.__version__)`

File: autosklearn/smbo.py (hunk 2 above)
Comment (mfeurer) [reply autosklearn/smbo.py:369]: If it's not used you can remove it.

File: autosklearn/evaluation/train_evaluator.py (hunk 3 above)
Comment (mfeurer) [reply autosklearn/evaluation/train_evaluator.py:575]: Yes, you can remove it if it's not used.

File: autosklearn/__init__.py (hunk 1 above)
Comment (gui-miotto) [reply autosklearn/__init__.py:6]: ok

File: autosklearn/smbo.py (hunk 2 above)
Comment (gui-miotto) [reply autosklearn/smbo.py:369]: removed

File: autosklearn/evaluation/train_evaluator.py (hunk 3 above)
Comment (gui-miotto) [reply autosklearn/evaluation/train_evaluator.py:575]: removed

**Comment by gui-miotto** [comment]
@mfeurer : ready for review

**Pull request merged**